### PR TITLE
refactor: convert C++ enums to C++11 enum classes

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Use C++11 enum classes for better type safety.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

**Comment by nornagon** [comment]
looks like the build is failing:
```
../../electron/atom/browser/api/atom_api_web_contents.cc:1704:20: error: use of undeclared identifier 'BACKGROUND_PAGE'
  if (GetType() != BACKGROUND_PAGE) {
                   ^
```

**Comment by miniak** [comment]
> looks like the build is failing:
> 
> ```
> ../../electron/atom/browser/api/atom_api_web_contents.cc:1704:20: error: use of undeclared identifier 'BACKGROUND_PAGE'
>   if (GetType() != BACKGROUND_PAGE) {
>                    ^
> ```

missed this one, the code is not building on Mac. fixed.

File: atom/common/node_bindings.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace atom {
 
 class NodeBindings {
  public:
-  enum BrowserEnvironment {
+  enum class BrowserEnvironment {
```
Comment (deepak1556) [inline atom/common/node_bindings.h]: Maybe just `BrowserEnvironment`=>`Environment` is better

Comment (miniak) [reply atom/common/node_bindings.h]: done

Comment (miniak) [reply atom/common/node_bindings.h]: actually, it might get confused with `base::Environment`, let's keep it `BrowserEnvironment`

Comment (deepak1556) [reply atom/common/node_bindings.h]: I don't think it should, when they are being qualified by namespace. `BrowserEnvironment::Renderer`, `BrowserEnvironment::Worker` doesn't read correct comparatively. I would have suggested `NodeEnvironment` but that is more likely to be confused with `node::Environment`.

**Comment by miniak** [comment]
@zcbenz done

**Comment by codebytere** [comment]
@miniak
```
../../electron\atom/browser/net/asar/url_request_asar_job.h(70,5): error: expected identifier
    ERROR,
    ^
..\..\..\..\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\wingdi.h(118,29): note: expanded from macro 'ERROR'
#define ERROR               0
                            ^
In file included from ../../electron/atom/browser/api/atom_api_protocol.cc:10:
In file included from ../../electron\atom/browser/net/url_request_async_asar_job.h:10:
../../electron\atom/browser/net/asar/url_request_asar_job.h(114,28): error: expected unqualified-id
  JobType type_ = JobType::ERROR;
                           ^
..\..\..\..\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\wingdi.h(118,29): note: expanded from macro 'ERROR'
#define ERROR               0
                            ^
2 errors generated.
[16939/18125] CXX obj/electron/electron_lib/atom_api_dialog.obj
FAILED: obj/electron/electron_lib/atom_api_dialog.obj 
```

**Comment by miniak** [comment]
@codebytere fixing these. I don't have a Windows build available at this time, so I am relying on the CI.

**Pull request merged**